src/main/java/catan/services/impl/PlayServiceImpl.java

```diff
@@ -31,7 +29,108 @@
+        if (user == null) {
+            log.debug("User should not be empty");
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+
+        int edgeId;
+        try {
+            if (edgeIdString == null || edgeIdString.trim().length() == 0) {
+                log.debug("Cannot build road on empty edgeId");
+                throw new PlayException(ERROR_CODE_ERROR);
+            }
+
+            edgeId = Integer.parseInt(edgeIdString);
+        } catch (Exception e) {
+            log.debug("Cannot convert edgeId to integer value");
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+        GameBean game = gameUtil.getGameById(gameIdString, ERROR_CODE_ERROR);
+        if (game.getStatus() != GameStatus.PLAYING) {
+            log.debug("Cannot build road in not playing game");
+            throw new GameException(ERROR_CODE_ERROR);
+        }
+
+        //TODO: move to util method and refactor all other places
+        GameUserBean gameUserBean = null;
+        for (GameUserBean gameUser : game.getGameUsers()) {
+            if (gameUser.getUser().equals(user)) {
+                gameUserBean = gameUser;
+                break;
+            }
+        }
+
+        if (gameUserBean == null) {
+            log.debug("User is not joined to game id specified %s", gameIdString);
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+        EdgeBean edgeToBuildOn = null;
+        for (EdgeBean edge : game.getEdges()) {
+            if (edge.getId() == edgeId) {
+                edgeToBuildOn = edge;
+            }
+        }
+
+        if (edgeToBuildOn == null) {
+            log.debug("Cannot build road on edgeId {} that does not belong to game {}", edgeId, gameIdString);
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
 
+        if (edgeToBuildOn.getBuilding() != null) {
+            log.debug("Cannot build road on this edge as it already has building on it");
+            throw new PlayException(ERROR_CODE_ERROR);
+        }
+
+        boolean nearNeighbourRoad = false;
+        boolean nearNeighbourSettlement = false;
+
+        for (NodeBean node : edgeToBuildOn.getNodes().all()) {
+            if (node == null) {
+                continue;
+            }
+
+            if (node.getBuilding() == null) {
+                for (EdgeBean neighbourEdge : node.getEdges().all()) {
+                    if (neighbourEdge == null || neighbourEdge.equals(edgeToBuildOn)) {
+                        continue;
+                    }
+
+                    if (neighbourEdge.getBuilding() != null && neighbourEdge.getBuilding().getBuildingOwner().getUser().equals(user)) {
+                        nearNeighbourRoad = true;
+                    }
+                }
+            } else {
+                if (node.getBuilding().getBuildingOwner().getUser().equals(user)) {
+                    nearNeighbourSettlement = true;
+                } else {
```

Comment:
should be discussed. i think that you can build your road close to opponents' settlement in case if you have contiguous road from another side of road that you want to build.